plugin/src/main/java/org/opensearch/ml/action/deploy/TransportDeployModelAction.java

```diff
@@ -114,6 +118,7 @@ public TransportDeployModelAction(
         this.mlModelManager = mlModelManager;
         this.mlStats = mlStats;
         this.modelAccessControlHelper = modelAccessControlHelper;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
```

Comment:
Need to add setting update consumer. Check line 125 of this class

Reply:
Sorry, after reading the whole code changes, I know why you write like this way. I guess you are trying to avoid some duplicate code by creating a new `MLFeatureEnabledSetting` class.

Reply:
Yes you are right. More new feature flags can be added into this class in the future releases.